projects/netgrif-components-core/src/lib/forms/change-password/abstract-change-password.component.ts

```diff
@@ -0,0 +1,107 @@
+import {TranslateService} from '@ngx-translate/core';
+import {Component, EventEmitter, OnDestroy, Output} from '@angular/core';
+import {FormSubmitEvent, HasForm} from "../has-form";
+import {LoadingEmitter} from "../../utility/loading-emitter";
+import {take} from "rxjs/operators";
+import {UserChangePasswordRequest} from "../../authentication/profile/models/user-change-password-request";
+import {ProfileService} from "../../authentication/profile/services/profile.service";
+import {UserService} from "../../user/services/user.service";
+import {encodeBase64} from '../../utility/base64';
+
+@Component({
+    selector: 'ncc-abstract-forgotten-password',
+    template: ''
+})
+export abstract class AbstractChangePasswordComponent implements HasForm, OnDestroy {
+
+    protected readonly MIN_PASSWORD_LENGTH = 8;
+
+    public rootFormGroup: FormGroup;
+    public hideOldPassword: boolean;
+
+    public hidePassword: boolean;
+    public hideRepeatPassword: boolean;
+
+    @Output() public formSubmit: EventEmitter<FormSubmitEvent>;
+    @Output() public changePassword: EventEmitter<MessageResource>;
+
+    public loadingSubmit: LoadingEmitter;
+
+    protected constructor(protected formBuilder: FormBuilder,
+                          protected profileService: ProfileService,
+                          protected user: UserService,
+                          protected _log: LoggerService,
+                          protected _translate: TranslateService) {
+        this.hidePassword = true;
+        this.hideOldPassword = true;
+        this.hideRepeatPassword = true;
+        this.formSubmit = new EventEmitter<FormSubmitEvent>();
+        this.changePassword = new EventEmitter<MessageResource>();
+        this.loadingSubmit = new LoadingEmitter(false);
+        this.rootFormGroup = formBuilder.group({
+            oldPassword: ['', [Validators.required, Validators.minLength(1)]],
+            password: ['', [Validators.required, Validators.minLength(this.MIN_PASSWORD_LENGTH)]],
+            confirmPassword: ['', [Validators.required, Validators.minLength(this.MIN_PASSWORD_LENGTH)]]
+        }, {validator: passwordValidator});
+    }
+
+
+    public isFieldValid(formControlName: string): boolean {
+        return this.rootFormGroup.get(formControlName).valid;
+    }
+
+    protected createRequestBody(): UserChangePasswordRequest {
+        return {
+            login: this.user.user.email,
+            password: encodeBase64(this.rootFormGroup.controls['oldPassword'].value),
+            newPassword: encodeBase64(this.rootFormGroup.controls['password'].value)
+        };
+    }
+
+    public getErrorMessage(formControlName: string): string {
+        const errors = this.rootFormGroup.get(formControlName).errors;
+        if (errors === null) {
+            return;
+        }
+        switch (Object.keys(errors)[0]) {
+            case 'mismatchedPassword':
+                return this._translate.instant('forms.register.passwordsMustMatch');
```

Comment:
where are these texts used? If they are only displayed, it might be better to return the translation codes only and use the translate pipe to translate them on display. Such an implementation would work event if the display language would change when the text is displayed

Reply:
New Task....